src/pages/storage/forms/StoragePoolFormMain.tsx

```diff
@@ -124,22 +130,26 @@ const StoragePoolFormMain: FC<Props> = ({ formik }) => {
                 disabled={formik.values.driver === dirDriver}
               />
             )}
-          {!isPureDriver && !isPowerFlexDriver && (
-            <Input
-              {...getFormProps("source")}
-              type="text"
-              disabled={
-                formik.values.driver === btrfsDriver ||
-                !formik.values.isCreating
-              }
-              help={
-                formik.values.isCreating
-                  ? getSourceHelpForDriver(formik.values.driver)
-                  : "Source can't be changed"
-              }
-              label="Source"
-            />
-          )}
+          {!isPureDriver &&
+            !isPowerFlexDriver &&
+            (canClusterSource && isClusteredServer(settings) ? (
```

Comment:
```suggestion
            (hasClusterWideSource || !isClusteredServer(settings) ? (
```
Shouldn't this be negated, so we avoid going into the cluster specific branch in case of a non-clustered server?